### DISCUSSION TITLE
support for response code 202 in HTTP source plugin

### DISCUSSION THREAD

**Discussion by mp3monster** [fluent/fluent-bit#7599] [Ideas]
If Fluent Bit was to accept and then possibly discard a log message received, would it not be correct to return a 202 response code?

https://datatracker.ietf.org/doc/html/rfc7231#section-6.3.3

**Comment by patrick-stephens** [comment]
Do you mean discard at the input level (I'm not sure how it could without knowing the HTTP input source but presumably it could with buffers maxed or something?) or conceptually in that it never makes it out of the pipeline?

I guess the issue is that discarding is downstream of the input and potentially you can have multiple filters/outputs either accepting or discarding so what do you do in that case if one pipeline allows it but the other discards it?

**Comment by mp3monster** [comment]
I was thinking if you log event pipeline looked like :   HTTP Source --> filter --> xyz target plugin. The HTTP Source would be legitimate in having a 202  for the http_response_code attribute. Currently, 202 is not allowed. So the change being suggested would be allow the plugin configuration to return 202
________________________________
